GeeksCoreLibrary/Modules/Templates/Services/PagesService.cs

```diff
@@ -248,6 +249,12 @@ public async Task<PageViewModel> CreatePageViewModelAsync(List<string> externalC
                         continue;
                     }
 
+                    // Validate the template regex, if it has one.
+                    if (!String.IsNullOrWhiteSpace(template.UrlRegex) && !String.IsNullOrWhiteSpace(currentUrl) && !Regex.IsMatch(currentUrl, template.UrlRegex))
```

Comment:
Kun je aub een timeout toevoegen aan de `Regex.IsMatch`, zoals ik laatst had aangegeven in .NET overleg?

Reply:
Ik heb een timeout toegevoegd.